public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocation.utils.ts

```diff
@@ -0,0 +1,65 @@
+import { StorageLocation, LocationType, S3Location } from '../StorageLocations.types';
+import { AddStorageLocationFormProps } from './AddStorageLocationModal.types';
+
+export const toStorageLocation = (values: AddStorageLocationFormProps): StorageLocation => {
+  const { name, description, type, endpoint, client, server, accessKey, secretKey } = values;
+  const result: Partial<StorageLocation> = { name, description };
+
+  switch (type) {
```

Comment:
Switch on type is a code smell, see [this](https://makolyte.com/refactoring-the-switch-statement-code-smell), [this](https://wiki.c2.com/?SwitchStatementsSmell) and [this](https://refactoring.guru/smells/switch-statements)

Reply:
Switched to map